flow-typed/@react-navigation/material-top-tabs_v5.x.x.js

```diff
@@ -840,11 +840,13 @@ declare module '@react-navigation/material-top-tabs' {
     >>,
     +setOptions: (options: $Shape<ScreenOptions>) => void,
     +setParams: (
-      params: $If<
-        $IsUndefined<$ElementType<ParamList, RouteName>>,
-        empty,
-        $Shape<$NonMaybeType<$ElementType<ParamList, RouteName>>>,
-      >,
+      // We've edited this to be less complicated, so Flow in types-first
```

Comment:
We should really deduplicate this stuff; we've found out that this
    should be possible with the .js.flow strategy Greg started using in
    007dea350.

Wow, I see, these files really have a ton in common!

They're all over 2000 lines long:
```
$ git ls flow-typed/ | xargs wc -l
…
  2101 flow-typed/@react-navigation/drawer_v5.x.x.js
  2046 flow-typed/@react-navigation/material-top-tabs_v5.x.x.js
  2185 flow-typed/@react-navigation/native_v5.x.x.js
  2137 flow-typed/@react-navigation/stack_v5.x.x.js
```

But the diffs are much smaller than that:
```
$ git di --stat upstream:flow-typed/@react-navigation/{material-top,bottom}-tabs_v5.x.x.js 
 ...rial-top-tabs_v5.x.x.js => bottom-tabs_v5.x.x.js} | 38 ++++++++++------------
 1 file changed, 18 insertions(+), 20 deletions(-)

$ git di --stat upstream:flow-typed/@react-navigation/{bottom-tabs,stack}_v5.x.x.js 
 .../{bottom-tabs_v5.x.x.js => stack_v5.x.x.js}       | 141 +++++++++++++++++----
 1 file changed, 118 insertions(+), 23 deletions(-)
```
(and similarly the others.)